### PR TITLE
[Github CI] Use the same python3 version in all Github Workflows

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR:
- Introduces a `python3-version` file storing the python3 version used in the repository
- Makes all Github Workflows use the python3 version specified in the `.python3-version` file.

### Motivation

Cleaner code, better reproducibility.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by modernplumbing — APPROVED** [review]

File: .python3-version
Comment (chouquette) [inline .python3-version:1]: Is it possible to move this file in the `github` folder? We have a pretty crammed root folder already IMO

File: .python3-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.9.5
```
Comment (chouetz) [inline .python3-version:1]: Could we use a more recent version? 3.9 end of life was in 2022

**Review by chouetz — APPROVED** [review]

File: .github/workflows/cws-btfhub-sync.yml [hunk 2]
```diff
@@ -26,7 +26,7 @@ jobs:
       - name: Install python
         uses: actions/setup-python@v4
         with:
-          python-version: '3.9'
+          python-version-file: '.python3-version'
```
Comment (paulcacheux) [inline .github/workflows/cws-btfhub-sync.yml:29]: Here is an example run for this job: https://github.com/DataDog/datadog-agent/actions/runs/6742552121/job/18328848192
TLDR there is no 3.9.5 for ubuntu latest (22.04).

**Review by paulcacheux — CHANGES_REQUESTED** [review]

File: .python3-version
Comment (amenasria) [reply .python3-version:1]: In my mind the .python3-version has a bigger purpose than just the github workflows. Basically my goal for introducing this file is to use the same version for local setup and for buildimages.

Would it be fine keeping it in the root folder for this purpose ?

File: .python3-version (hunk 1 above)
Comment (amenasria) [reply .python3-version:1]: Agreed that we should bump it ! I'll do it in another PR though !

File: .github/workflows/cws-btfhub-sync.yml (hunk 2 above)
Comment (amenasria) [reply .github/workflows/cws-btfhub-sync.yml:29]: My bad ! Just found the available versions for [ubuntu 22.04](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#python) and [ubuntu 20.04](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#python).

Bumping it to 3.9.18 !

File: .python3-version
Comment (chouquette) [reply .python3-version:1]: If you're planning some followup PRs that would use this file then not at all :)
If it's more "just in case" then I'm a bit more reluctant

**Review by lrbhatia — APPROVED** [review]

File: .python3-version
Comment (amenasria) [reply .python3-version:1]: Yup, it will be used in a followup PR ! That's a good point for the future though, it might be nice to find a way to lighten the number of files in the root folder. When I'm looking at big repo such as web-ui or dogweb they're still pretty crowded as well (even more than us) though :/
